Editor/MenuActions/Editors/NewPolyShape.cs

```diff
@@ -32,8 +33,52 @@ public override bool enabled
             get { return true; }
         }
 
+        bool CanCreateNewPolyShape()
+        {
+            //If inspector is locked we cannot create new PolyShape.
+            //First created inspector seems to hold a specific semantic where
+            //if not unlocked no matter how many inspectors are present they will
+            //not allow the creation of new PolyShape.
+            var inspectorType = typeof(Editor).Assembly.GetType("UnityEditor.InspectorWindow");
```

Comment:
This looks good for `2018.{3,4}`, though I would still use the original `InspectorWindow.GetWindows` if Unity is 2019.1 or up